src/StackExchange.Redis/Format.cs

```diff
@@ -355,5 +355,88 @@ internal static unsafe string GetString(ReadOnlySpan<byte> span)
                 return Encoding.UTF8.GetString(ptr, span.Length);
             }
         }
+
+        [DoesNotReturn]
+        private static void ThrowFormatFailed() => throw new InvalidOperationException("TryFormat failed");
+
+        internal const int
+            MaxInt32TextLen = 11, // -2,147,483,648 (not including the commas)
+            MaxInt64TextLen = 20; // -9,223,372,036,854,775,808 (not including the commas)
+
+        internal static int MeasureDouble(double value)
+        {
+            if (double.IsInfinity(value)) return 4; // +inf / -inf
+            var s = value.ToString("G17", NumberFormatInfo.InvariantInfo); // this looks inefficient, but is how Utf8Formatter works too, just: more direct
+            return s.Length;
+        }
+
+        internal static int FormatDouble(double value, Span<byte> destination)
+        {
+            if (double.IsInfinity(value))
+            {
+                if (double.IsPositiveInfinity(value))
+                {
+                    if (!"+inf"u8.TryCopyTo(destination))
+                    {
+                        ThrowFormatFailed();
+                    }
+                    return 4;
+                }
+                if (double.IsNegativeInfinity(value))
```

Comment:
Can this just be a cheaper else, given the context?

Reply:
possibly; I was mirroring existing code, but yes: I can't think of a third kind of infinity :)

Reply:
tweaked; please recheck

Reply:
Looking good 👍

Reply:
two infinities, and beyond! 🚀